src/data/finsupp/witnesses.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.finsupp.basic
+
+/-!
+#  Witnesses for finitely supported functions
+
+Let `α N` be two Types, and assume that `N` has a `0` and let `f g : α →₀ N` be finitely supported
+functions.
+
+##  Main definitions
+
+*  `finsupp.diff f g : finset α`, the finite subset of `α` where `f` and `g` differ,
+*  `finsupp.wit f g : α`, [assuming that `a` is non-empty and `N` is linearly ordered]
+   the minimum of `finsupp.diff f g` (or a random element of `α` if
+   `finsupp.diff f g` is empty, i.e. if `f = g`).
+
+In the case in which `N` is an additive group, `finsupp.diff f g` coincides with
+`finsupp.support (f - g)`.
+-/
+
+variables {α N : Type*}
+
+namespace finsupp
+variables [decidable_eq α] [decidable_eq N]
+
+section diff
+
+section N_has_zero
+variables [has_zero N] {f g : α →₀ N}
+
+/--  Given two finitely supported functions `f g : α →₀ N`, `finsupp.diff f g` is the `finset`
+where `f` and `g` differ. This generalizes `(f - g).support` to situations without subtraction. -/
+def diff (f g : α →₀ N) : finset α :=
```

Comment:
One other possibility would be to call this `ne_locus` to match `ring_hom.eq_locus` and similarly named definitions. Let's see if any other reviewers have opinions